parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java

```diff
@@ -678,6 +656,46 @@ public void testUnionWithSingleNonNullType() throws Exception {
     assertEquals(str("theValue"), nextRecord.get("value"));
   }
 
+  @Test
+  public void testDuplicatedValuesWithDictionary() throws Exception {
```

Comment:
This test fails when the fix is not applied, right?

Reply:
Yes, it should fail. Actually it is the example code supplied by the reporter, I just made a unit test from it. The test failed without your fix, and was green with it.